### PR TITLE
Fix fee capitalization

### PR DESCRIPTION
**Pull request by 34x4p08**
- possibility to set multiple minters for USDP
- no fee capitalization
- no COL token related logic
- `tokenDebts` fix

### CODE REVIEW

File: contracts/vault-managers/CDPManager01.sol [hunk 1]
```diff
@@ -222,8 +252,9 @@ contract CDPManager01 is ReentrancyGuard {
 
     // decreases debt
     function _repay(address asset, address owner, uint usdpAmount) internal {
-        uint fee = vault.calculateFee(asset, owner, usdpAmount);
+        uint fee = vault.checkpointFee(asset, owner);
```
Comment (Eenae) [inline contracts/vault-managers/CDPManager01.sol:255]: All the fee is paid at once. That also means that the `owner` may be required to have a significant amount of USDP in addition to `usdpAmount`.
Alternatively, we may pay the fee first, up to `usdpAmount`. And `repay` the remaining `usdpAmount - fee`, if any.

File: contracts/Vault.sol [hunk 2]
```diff
@@ -93,10 +89,7 @@ contract Vault is Auth {
      **/
     function update(address asset, address user) public hasVaultAccess notLiquidating(asset, user) {
 
-        // calculate fee using stored stability fee
-        uint debtWithFee = getTotalDebt(asset, user);
-        tokenDebts[asset] = tokenDebts[asset].sub(debts[asset][user]).add(debtWithFee);
-        debts[asset][user] = debtWithFee;
+        accumulatedStabilityFee[asset][user] = getFee(asset, user);
```
[the remark this answers is not in the document]
Comment (34x4p08) [reply contracts/Vault.sol]: applied

File: contracts/vault-managers/CDPManager01.sol (hunk 1 above)
Comment (34x4p08) [reply contracts/vault-managers/CDPManager01.sol]: applied

File: contracts/Vault.sol [hunk 3]
```diff
@@ -299,9 +288,7 @@ contract Vault is Auth {
         address asset,
         address positionOwner,
         uint mainAssetToLiquidator,
-        uint colToLiquidator,
         uint mainAssetToPositionOwner,
-        uint colToPositionOwner,
         uint repayment,
```
Comment (Eenae) [inline contracts/Vault.sol:292]: During liquidation, accrued stability fee is included into `repayment` and gets burnt, not transferred to the foundation as usual. Is it the right behavior?

File: contracts/Vault.sol [hunk 4]
```diff
@@ -236,14 +209,15 @@ contract Vault is Auth {
     external
     hasVaultAccess
     notLiquidating(asset, user)
+    checkpointFee(asset, user)
     returns(uint)
     {
         uint debt = debts[asset][user];
         debts[asset][user] = debt.sub(amount);
         tokenDebts[asset] = tokenDebts[asset].sub(amount);
```
[the remark this answers is not in the document]
Comment (34x4p08) [reply contracts/Vault.sol:217]: Yes

**Pull request merged**